src/sider/ftrl/saksbehandling/stegKomponenter/vurderingBestemmelse/vurderingBestemmelse.tsx

```diff
@@ -97,10 +98,31 @@ export const VurderingBestemmelse = ({ bekreft, tilbake, aktivtSteg, oppdaterSta
     const valgtBestemmelseMedVilkårOgBegrunnelser = støttedeBestemmelser.find(
       (element) => element.bestemmelse === valgtBestemmelse
     );
+
     const alleVilkårHarSvarJaOgValgtBegrunnelse = valgtBestemmelseMedVilkårOgBegrunnelser?.vilkårOgBegrunnelser.every(
-      (element) =>
-        valgteVilkår.get(element.vilkår) === SANN &&
-        (Utils._isEmpty(element.muligeBegrunnelser) ? true : valgteBegrunnelser.get(`${element.vilkår}_begrunnelser`))
+      (element) => {
+        const valgtBegrunnelseForVilkår = valgteBegrunnelser.get(`${element.vilkår}_begrunnelser`);
+        const begrunnelseInneholderFritekst = KV.termFraNestedKTObject(
+          begrunnelseKodeverk,
+          valgtBegrunnelseForVilkår?.begrunnelseKode
+        )?.includes("(fritekst)");
+
+        const harVilkår = valgteVilkår.get(element.vilkår) === SANN;
+        const ingenMuligeBegrunnelser = Utils._isEmpty(element.muligeBegrunnelser);
+        const harValgtFritekstBegrunnelse = begrunnelseInneholderFritekst;
+        const valgtBegrunnelseFinnes = valgtBegrunnelseForVilkår;
+        const harStrengInnholdIFritekst = harStrengInnhold(valgtBegrunnelseForVilkår?.begrunnelseFritekst ?? "");
+
+        if (ingenMuligeBegrunnelser) {
+          return harVilkår;
+        }
+
+        if (!harValgtFritekstBegrunnelse) {
+          return harVilkår && valgtBegrunnelseFinnes;
+        }
+
+        return harVilkår && harStrengInnholdIFritekst;
+      }
```

Comment:
```suggestion
        const harVilkår = valgteVilkår.get(element.vilkår) === SANN;
        const valgtBegrunnelseForVilkår = valgteBegrunnelser.get(`${element.vilkår}_begrunnelser`);
        const begrunnelseInneholderFritekst = KV.termFraNestedKTObject(
          begrunnelseKodeverk,
          valgtBegrunnelseForVilkår?.begrunnelseKode
        )?.includes("(fritekst)");

        if (Utils._isEmpty(element.muligeBegrunnelser)) {
          return harVilkår;
        }

        if (!begrunnelseInneholderFritekst) {
          return harVilkår && valgtBegrunnelseForVilkår;
        }

        return harVilkår && harStrengInnhold(valgtBegrunnelseForVilkår?.begrunnelseFritekst ?? "");
```

Dette var en alternativ måte å skrive det på med mindre hjelpevariabler. 
Valgte å bruke flere variabler slik at hovedflyten (nederst ved if-ene) ble lettere å lese. 
Hvis du er _veldig_ uenig ved bruk av variabler som ble comitted, kan vi selvsagt gjøre det som dette forslaget :smiley_cat: 
@janitamarielle

Reply:
Dette ser flott ut! 👏

Reply:
Med eller uten ekstra variabler?

Reply:
Slik du skrev det i eksemplet over ☝️